applications/minotari_app_utilities/src/parse_miner_input.rs

```diff
@@ -0,0 +1,195 @@
+// Copyright 2021. The Tari Project
+//
+// Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+// following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+// disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+// following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+// products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use std::{net::SocketAddr, str::FromStr};
+
+use dialoguer::Input as InputPrompt;
+use minotari_app_grpc::{
+    authentication::ClientAuthenticationInterceptor,
+    tari_rpc::{base_node_client::BaseNodeClient, Block, NewBlockTemplate, NewBlockTemplateRequest},
+};
+use tari_common::configuration::{
+    bootstrap::{grpc_default_port, ApplicationType},
+    Network,
+};
+use tari_common_types::tari_address::TariAddress;
+use tari_comms::{multiaddr::Multiaddr, utils::multiaddr::multiaddr_to_socketaddr};
+use thiserror::Error;
+use tonic::{codegen::InterceptedService, transport::Channel, Code};
+
+/// Error parsing input
+#[derive(Debug, Error)]
+pub enum ParseInputError {
+    #[error("Could not convert data:{0}")]
+    WalletPaymentAddress(String),
+    #[error("Could not convert data:{0}")]
+    BaseNodeSocketAddress(String),
+}
+
+/// Read base_node_socket_address arg or prompt for input
+pub fn base_node_socket_address(
```

Comment:
a note here, we need to decide if this is fine or not.

Currently, the miner chooses the default address as `127.0.0.1:{Network port}`
If you want a non interactive deploy, you need to set the correct default address in the config. If not the miner will block on user prompt.